src/client/resources/conversions.ts

```diff
@@ -155,13 +154,71 @@ export function createMigPlan(
         name: storageObj,
         namespace,
       },
-      migAssetCollectionRef: {
-        name: assetObj,
+    },
+  };
+}
+
+export function updateMigPlanFromValues(
+  migPlan: any,
+  planValues: any,
+) {
+  const updatedSpec = Object.assign({}, migPlan.spec);
+
+  updatedSpec.migStorageRef = {
+    name: planValues.selectedStorage,
+    namespace: migPlan.metadata.namespace,
+  };
+
+  updatedSpec.persistentVolumes = updatedSpec.persistentVolumes.map(v => {
+    const userPv = planValues.persistentVolumes.find(upv => upv.name === v.name);
+    v.action = userPv.type;
+    return v;
+  });
+
+  return {
+    apiVersion: 'migration.openshift.io/v1alpha1',
+    kind: 'MigPlan',
+    metadata: migPlan.metadata,
+    spec: updatedSpec,
+  };
+}
+
+export function createMigPlanNoStorage(
+  name: string,
+  namespace: string,
+  sourceClusterObj: any,
+  destinationClusterObj: any,
+  namespaces: string[],
+) {
+  return {
+    apiVersion: 'migration.openshift.io/v1alpha1',
+    kind: 'MigPlan',
+    metadata: {
+      name,
+      namespace,
+    },
+    spec: {
+      srcMigClusterRef: {
+        name: sourceClusterObj,
         namespace,
       },
+      destMigClusterRef: {
+        name: destinationClusterObj,
+        namespace,
+      },
+      namespaces,
+      //////////////////////////////////////////////////////////////////////////
+      // TODO: Need to rip this out once the controller allows for creation
```

Comment:
This is required for now, but we'll want to rip this out and test with an updated controller here, so the method continues to make sense.

[ISSUE]

Reply:
https://github.com/fusor/mig-ui/issues/227